AdfsEventsModule.psm1

```diff
@@ -4,6 +4,130 @@
+	$OldSeSecPriv = $OldSeSecPriv.Line
+	$NewSeSecPriv = $OldSeSecPriv  + ",*" + $SvcAcctSID
+	(gc $SecTempPath).replace($OldSeSecPriv,$NewSeSecPriv) | Out-File -Filepath $SecTempPath 
+	secedit /configure /db c:\windows\security\local.sdb /cfg $SecTempPath /areas SECURITYPOLICY 
+	$RM = rm -force $SecTempPath -confirm:$false -ErrorAction SilentlyContinue
+	gpupdate /force
+
+	#Enable ADFS Tracing log
+	$ADFSTraceLogName = "AD FS Tracing/Debug"
+	$ADFSTraceLog = New-Object System.Diagnostics.Eventing.Reader.EventlogConfiguration $ADFSTraceLogName
+	if($ADFSTraceLog.IsEnabled -ne $true)
+	{
+		$ADFSTraceLog.IsEnabled = $true
+		$ADFSTraceLog.SaveChanges()
+	}
+
+	#Enable security auditing from ADFS
+	switch ($OSVersion.Buildnumber)
+	{
+		'6000'{Add-PsSnapin Microsoft.Adfs.Powershell -ErrorAction SilentlyContinue}
+		'6001'{Add-PsSnapin Microsoft.Adfs.Powershell -ErrorAction SilentlyContinue}
+		'6002'{Add-PsSnapin Microsoft.Adfs.Powershell -ErrorAction SilentlyContinue}
+		'7600'{Add-PsSnapin Microsoft.Adfs.Powershell -ErrorAction SilentlyContinue}
+		'7601'{Add-PsSnapin Microsoft.Adfs.Powershell -ErrorAction SilentlyContinue}
+		'9200'{Import-Module ADFS -ErrorAction SilentlyContinue}
+		'9600'{Import-Module ADFS -ErrorAction SilentlyContinue}
+		'14393'{Import-Module ADFS -ErrorAction SilentlyContinue}
+	}
+
+    $SyncProps = Get-ADFSSyncProperties
+    if ($SyncProps.Role -ne 'SecondaryComputer') 
+    {
+        Set-ADFSProperties -LogLevel  @( "FailureAudits", "SuccessAudits", "Warnings", "Verbose", "Errors", "Information")
+    }
+	auditpol.exe /set /subcategory:"Application Generated" /failure:enable /success:enable
+
+	Write-host "ADFS auditing is now enabled."
+}
+
+#function to enable ADFS auditing. Takes in parameters of server names
+function Disable-ADFSAuditing
+{
+	<#
+    .SYNOPSIS
+    This script disables ADFS verbose related events from the security, admin, and debug logs.
+
+    .DESCRIPTION
+	To track ADFS authentication processing there are multiple items which must be enabled on the ADFS server(s). This function provides 
+    automation for disabling those items so that event logs do not fill up.
+    Note that this function can only run on the local server-it cannot be run against a set of servers.
+	
+	EXAMPLE
+    	Disable-ADFSAuditing
+	#>
+	
+	#Disable ADFS Tracing log
+	$ADFSTraceLogName = "AD FS Tracing/Debug"
+	$ADFSTraceLog = New-Object System.Diagnostics.Eventing.Reader.EventlogConfiguration $ADFSTraceLogName
+	if ($ADFSTraceLog.IsEnabled -ne $false)
+	{
+		$ADFSTraceLog.IsEnabled = $false
+		$ADFSTraceLog.SaveChanges()
+	}
+
+	#Disable security auditing from ADFS
+	$cs = get-wmiobject -class win32_computersystem -ComputerName "localhost"
+	$DomainRole = $cs.domainrole
+	$OSVersion = gwmi win32_operatingsystem
+	[int]$BuildNumber = $OSVersion.BuildNumber 
+	switch ($OSVersion.Buildnumber)
+	{
+		'6000'{Add-PsSnapin Microsoft.Adfs.Powershell -ErrorAction SilentlyContinue}

```

Comment:
Are you sure you want to silentlycontinue if the module can't be loaded?  Isn't that a fatal error for this script?

Reply:
If we fail to load the PowerShell, then the only thing we can't enable is the log level in Adfs Properties. Do you think there are error scenarios where the module might fail to load when we call it, but it is already loaded successfully? In that case, failing silently will lead to a success